src/frontend/hooks/server/track.ts

```diff
@@ -39,3 +40,12 @@ export function useDeleteTrackMutation() {
     docType: 'track',
   });
 }
+
+export function useGetPresetFromTrack(track: Track) {
```

Comment:
one of the principles that we are trying to follow is that functions should be agnostic and take the minimal amount of props. In this case, the prop should be `trackPresetRefId:string|undefined`. In otherwords it does not need the entire track object if it is only using the `presetRef` Id